packages/amplify-e2e-tests/src/__tests__/pinpoint/notifications-pinpoint-config-util.ts

```diff
@@ -0,0 +1,113 @@
+import {
+  addNotificationChannel,
+  amplifyPullNonInteractive,
+  amplifyPushAuth,
+  createNewProjectDir,
+  deleteProjectDir,
+  getAmplifyFlutterConfig,
+  getAppId,
+  removeAllNotificationChannel,
+} from '@aws-amplify/amplify-e2e-core';
+import { $TSAny, JSONUtilities } from 'amplify-cli-core';
+import * as fs from 'fs-extra';
+import * as path from 'path';
+import { getShortId } from '../../import-helpers';
+import { getAWSExports } from '../../schema-api-directives';
+
+const pinpointSettings = { resourceName: `notifications${getShortId()}` };
+
+export const runPinpointConfigTest = async (
+  projectRoot: string,
+  envName: string,
+  frontendConfig: { frontend: string; config?: { ResDir?: string } },
+  validate: (projectRoot: string, channels: string[]) => void,
+): Promise<void> => {
+  const appId = getAppId(projectRoot);
+  expect(appId).toBeDefined();
+
+  await addNotificationChannel(projectRoot, pinpointSettings, 'In-App Messaging');
+  await amplifyPushAuth(projectRoot);
+  validate(projectRoot, ['In-App Messaging']);
+
+  const secondRoot = await createNewProjectDir('notifications-pinpoint-pull');
+  try {
+    await amplifyPullNonInteractive(secondRoot, {
+      appId,
+      frontend: frontendConfig,
+      envName,
+    });
+    validate(secondRoot, ['In-App Messaging']);
+  } finally {
+    deleteProjectDir(secondRoot);
+  }
+
+  await addNotificationChannel(projectRoot, pinpointSettings, 'SMS', true, true);
+  validate(projectRoot, ['In-App Messaging', 'SMS']);
+
+  await removeAllNotificationChannel(projectRoot);
+  validate(projectRoot, []);
+};
+
+export const iosValidate = (projectRoot: string, channels: string[]): void => {
```

Comment:
a lot of this seems the same as the `analytics-pinpoint-config-util`. Any opportunity to share functions between the two?

Reply:
the difference is in the `validateNativeJsonConfig` which these functions eventually call and I thought is overkill to pass this as parameter 4 levels down the stack